diff --git a/browser/installer/windows/nsis/installer.nsi b/browser/installer/windows/nsis/installer.nsi
index c93fad5eb12b..c97c83b3dc9a 100755
--- a/browser/installer/windows/nsis/installer.nsi
+++ b/browser/installer/windows/nsis/installer.nsi
@@ -353,6 +353,13 @@ Section "-Application" APP_IDX
   ${CopyFilesFromDir} "$EXEDIR\core" "$INSTDIR" \
                       "$(ERROR_CREATE_DIRECTORY_PREFIX)" \
                       "$(ERROR_CREATE_DIRECTORY_SUFFIX)"
+  
+  ; Create the portable userData (Profile) folder
+  CreateDirectory "$INSTDIR\userData"
+
+  ; Set permissions: Full Control for Users group
+  AccessControl::GrantOnFile "$INSTDIR\userData" "(BU)" "FullAccess"
+  Pop $0
 
   ; Register DLLs
   ; XXXrstrong - AccessibleMarshal.dll can be used by multiple applications but
